### PR TITLE
ncm-ccm: switch to 00-tqu and buildtools 1.52

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

**Comment by stdweird** [comment]
@jrha panlint fails with what i consider a bug in panlint

File: ncm-ccm/src/main/pan/components/ccm/schema.pan [hunk 1]
```diff
@@ -35,7 +30,7 @@ type kerberos_principal_string = string with {
         error("Not a valid principal " + principal);
     };
 
-    realm = pc_r[len-1];
+    realm = pc_r[len - 1];
```
Comment (ned21) [inline ncm-ccm/src/main/pan/components/ccm/schema.pan]: hmm.. that looks a bit odd. Normally languages (e.g. pep8) allow no spaces in this context don't they?

Comment (stdweird) [inline ncm-ccm/src/main/pan/components/ccm/schema.pan]: it looks a bit odd, but i'd rather have panlint complain about this then no panlint. feel free to open an issue 😉

Comment (jrha) [inline ncm-ccm/src/main/pan/components/ccm/schema.pan]: You can always choose to ignore panlint!

**Comment by jrha** [comment]
@stdweird panlint is failing to recognise the annotation because there is a space before the `{` which is a by-the-book interpretation of the Pan documentation. It's not flagging the problem in a helpful way though.
You could remove the space to get the tests passing again.

File: ncm-ccm/src/main/pan/components/ccm/schema.pan (hunk 1 above)
Comment (jrha) [inline ncm-ccm/src/main/pan/components/ccm/schema.pan]: Also, please don't just change code to please panlint, if you disagree with what it is saying please open an issue.

Comment (ned21) [inline ncm-ccm/src/main/pan/components/ccm/schema.pan]: Raised as https://github.com/quattor/pan/issues/143

**Comment by stdweird** [comment]
retest this please

**Comment by stdweird** [comment]
retest this please

**Comment by stdweird** [comment]
retest this please

**Comment by jrha** [comment]
@stdweird I've merged your panlint fix, so now you need to unfix your whitespace change :grin:

**Comment by stdweird** [comment]
retest this please

**Comment by stdweird** [comment]
@jrha hmmm, still buggy. i'll see what i can do before the release

File: ncm-ccm/src/main/pan/components/ccm/schema.pan [hunk 2]
```diff
@@ -84,8 +80,14 @@ type component_ccm = {
     'ca_file' ? string
     @{The directory containing accepted CA certificates when using the https protocol.}
     'ca_dir' ? string
-    @{Wether the profiles should be group-readable (value is the groupname). There is no default, and it is not allowed to set both C<group_readable> and enable C<world_readable>.}
-    'group_readable' ? string with if (path_exists('/software/components/accounts')) {is_user_or_group('group', SELF)} else {true}
+    @{Wether the profiles should be group-readable (value is the groupname).
```
Comment (jrha) [inline ncm-ccm/src/main/pan/components/ccm/schema.pan]: W**h**ether

**Comment by jrha** [comment]
Code looks fine, I'm not bothered about the panlint problems right now.

**Comment by stdweird** [comment]
@jrha typo fixed

**Pull request merged**